### PR TITLE
Revision

### PR DESCRIPTION
**Pull request by ermshaua**
This pull request contains the changes made for the paper revision. Specifically, I implemented

1. store functions for reference EBI / DrugBank data, to make preon more easy to use
2. rounding to edit distance (as a parameter)
3. informative user warnings if no match is found (as a parameter)

Tests and examples were checked and adapted accordingly.

### CODE REVIEW

File: preon/drug.py [hunk 1]
```diff
@@ -1,10 +1,56 @@
...
 import pandas as pd
 
 
+def _store_resource_file(file_path, resource_file_name):
+    '''
+    Internal helper function to store a file in the resource directory.
+
+    Parameters
+    -----------
+    :param file_path: the file that should be stored.
```
Comment (mbenary) [inline preon/drug.py]: More precise, maybe like: 

> Path to the local file which should be included in the preon instance.

File: preon/drug.py [hunk 2]
```diff
@@ -1,10 +1,56 @@
...
+
+    resource_file = os.path.join(resource_path, resource_file_name)
+    shutil.copy(file_path, resource_file)
+
+
+def store_ebi_drugs(file_path):
+    '''
+    Stores EBI drug names downloaded from https://www.ebi.ac.uk/chembl/g/#search_results/compounds
+    as a CSV file in the local resources folder.
+
```
Comment (mbenary) [inline preon/drug.py:38]: Why are you selecting columns from the ebi-file? That should be part of the help-description.

File: preon/drug.py [hunk 3]
```diff
@@ -1,10 +1,56 @@
...
+    Parameters
+    -----------
+    :param file_path: the file path at which the EBI compund file is located
+
+    Examples
+    -----------
+    >>> store_ebi_drugs(file_path="/Users/Username/Downloads/compounds.csv")
+    '''
+    columns = ["Name", "Synonyms", "ChEMBL ID"]
```
Comment (mbenary) [inline preon/drug.py]: Consider refactoring. The same column names are used in load_ebi_drugs. If there is a change later on, we have to fix it twice.

File: preon/drug.py [hunk 4]
```diff
@@ -18,36 +64,55 @@ def load_ebi_drugs(file_path=f"{ABS_PATH}/resources/ebi_drugs.csv"):
...
+        # check for nan/empty, add chembl id
+        if isinstance(row["ChEMBL ID"], str) and len(row["ChEMBL ID"]) > 0:
             drug_names.extend(names)
-            chembl_ids.extend([chembl_id] * len(names))
+            chembl_ids.extend([row["ChEMBL ID"]] * len(names))
 
     return drug_names, chembl_ids
 
 
+def store_drugbank_drugs(file_path):
```
Comment (mbenary) [inline preon/drug.py:101]: Feature request for later: Create a general `store_xxx` function, which has parameters for file-path and column-names.

**Review by mbenary — CHANGES_REQUESTED** [review]
Please follow up on the documentation. Otherwise, it looks good.

File: preon/drug.py (hunk 1 above)
Comment (ermshaua) [reply preon/drug.py]: Done.

File: preon/drug.py (hunk 2 above)
Comment (ermshaua) [reply preon/drug.py:38]: I'm checking if the local file contains the required columns for loading and parsing. If I don't do this, we potentially store files we cannot load, which is an inconsistent state we do not want to have. Added the help description.

File: preon/drug.py (hunk 3 above)
Comment (ermshaua) [reply preon/drug.py]: Done.

File: preon/drug.py (hunk 4 above)
Comment (ermshaua) [reply preon/drug.py:101]: Yeah, maybe. Currently, we only store files we can load (specific store and load functions). If we add this, we store files that we cannot load. This should be discussed some time. Note, however, that the user can load any file she wants and fit preon with it, independent of the file being a local resource or not!

**Comment by ermshaua** [comment]
@mbenary Thanks for your feedback. I added it and discussed some points (see above). I would merge the changes now, if that's fine with you.

**Pull request merged**